patches/0010-src-api-Make-sockets.c-compatible-with-Musl.patch

```diff
@@ -0,0 +1,84 @@
+From 70f95e9fc226f7f74cc40a207563ae3858f6de7d Mon Sep 17 00:00:00 2001
+Message-Id: <70f95e9fc226f7f74cc40a207563ae3858f6de7d.1668957138.git.razvand@unikraft.io>
+In-Reply-To: <e245cb8fe567f6988ca1a008f4c63cbf4027558f.1668957138.git.razvand@unikraft.io>
+References: <e245cb8fe567f6988ca1a008f4c63cbf4027558f.1668957138.git.razvand@unikraft.io>
+From: Razvan Deaconescu <razvand@unikraft.io>
+Date: Sun, 20 Nov 2022 17:11:56 +0200
+Subject: [PATCH 4/4] src/api: Make sockets.c compatible with Musl
+
+Remove functions, macros and variables from `src/api/sockets.c` to make
+it compatible with Musl.
+
+Signed-off-by: Razvan Deaconescu <razvand@unikraft.io>
+---
+ src/api/sockets.c | 16 ++--------------
+ 1 file changed, 2 insertions(+), 14 deletions(-)
+
+diff --git a/src/api/sockets.c b/src/api/sockets.c
+index 097a6a9..ed06ea3 100644
+--- a/src/api/sockets.c
++++ b/src/api/sockets.c
+@@ -39,6 +39,7 @@
+ 
+ #include <sys/time.h>
+ 
++#define _BSD_SOURCE
```

Comment:
I think `*_SOURCE` defines should go before any `#include`s, otherwise you risk having a inconsistent set of declarations in scope.

Reply:
Usually, I would agree, but considering that we want the define to affect only the content in the source file itself.

At the same time, not sure why it's placed before another include, but, if it works :shrug:

Reply:
No, these kind of defines are specifically for system header files to support different standards (think of "BSD"/"GNU"/POSIX/UNIX/etc.)

Reply:
I also just noticed that _BSD_SOURCE is deprecated in glibc:
```
/* _BSD_SOURCE and _SVID_SOURCE are deprecated aliases for
   _DEFAULT_SOURCE.  If _DEFAULT_SOURCE is present we do not
   issue a warning; the expectation is that the source is being
   transitioned to use the new macro.  */
```
from `/usr/include/features.h`

Reply:
Okay, makes sense. I was usually going for "the minimal affected code size", which kind of makes sense when patching external libraries.

Okay, @razvand move it up and see if it still works.